### PR TITLE
Refactor config.php

### LINKED ISSUE #154: Refactor config.php file
**Is your feature request related to a problem? Please describe.**
Instead of having a config.php file for env variables, we could have .ENV file.

**Describe the solution you'd like**
We could have a file like this, it'd be easier for new users and more simpler when the project grows.

TOKEN =
USERNAME = 

https://github.com/laravel/laravel/blob/8.x/.env.example

### PR DESCRIPTION
**Pull request by komen205**
## Description

Fixes #154

### Type of change

- [ X ] New feature (added a non-breaking change which adds functionality)

## How Has This Been Tested?

- [ X] Tested locally with a valid username
- [ X] Tested locally with an invalid username
- [ X] Ran tests with `composer test`
- [x] Added or updated test cases to test new features

## Checklist:

- [ X] I have checked to make sure no other [pull requests](https://github.com/DenverCoder1/github-readme-streak-stats/pulls?q=is%3Apr+sort%3Aupdated-desc+) are open for this issue
- [ X] The code is properly formatted and is consistent with the existing code style
- [ X] I have commented my code, particularly in hard-to-understand areas
- [ X] I have made corresponding changes to the documentation
- [ X] My changes generate no new warnings

## Screenshots

image

### CODE REVIEW

**Comment by DenverCoder1** [comment]
I'm getting this error when I try running it. I've tried composer update and all that.

```sh
[::1]:38294 [500]: GET / - Uncaught Error: Failed opening required '..\vendor\autoload.php' (include_path='.:')
in ..../github-readme-streak-stats/src/index.php:5
Stack trace:
#0 {main}
  thrown in ..../github-readme-streak-stats/src/index.php on line 5
```

**Comment by DenverCoder1** [comment]
I needed to use forward slash instead of backslash.

Maybe we can find a way that's compatible with both systems?

**Comment by komen205** [comment]
> I needed to use forward slash instead of backslash.
> 
> Maybe we can find a way that's compatible with both systems?

I think forward slash works on my system too, I will test it.

**Comment by komen205** [comment]
Fixed! @DenverCoder1

File: src/index.php [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
 require_once "stats.php";
 require_once "card.php";
 
-// load config if the file exists
-if (file_exists("config.php")) {
-    require_once "config.php";
-}
+// load .env
+
+$dotenv = \Dotenv\Dotenv::createImmutable(dirname(__DIR__,1));
```
Comment (DenverCoder1) [inline src/index.php:11]:
```suggestion
// load .env
$dotenv = \Dotenv\Dotenv::createImmutable(dirname(__DIR__, 1));
```

File: tests/StatsTest.php [hunk 2]
```diff
@@ -2,17 +2,20 @@
 use PHPUnit\Framework\TestCase;
 
 // load functions
+require_once dirname(__DIR__,1).'/vendor/autoload.php';
```
Comment (DenverCoder1) [inline tests/StatsTest.php:5]:
```suggestion
require_once dirname(__DIR__, 1) . '/vendor/autoload.php';
```

File: tests/StatsTest.php [hunk 3]
```diff
@@ -2,17 +2,20 @@
...
+// load .env
+$dotenv = \Dotenv\Dotenv::createImmutable(dirname(__DIR__,1));
+$dotenv->safeLoad();
+
+
 // if environment variables are not loaded, display error
-if (!getenv("TOKEN") || !getenv("USERNAME")) {
-    $message = file_exists("config.php")
+if (!$_SERVER["TOKEN"] || !$_SERVER["USERNAME"]) {
+    $message = file_exists(dirname(__DIR__.'.env',1))
```
Comment (DenverCoder1) [inline tests/StatsTest.php:15]:
```suggestion
    $message = file_exists(dirname(__DIR__ . '.env', 1))
```

File: tests/StatsTest.php [hunk 4]
```diff
@@ -2,17 +2,20 @@
...
 // load functions
+require_once dirname(__DIR__,1).'/vendor/autoload.php';
 require_once "src/stats.php";
 
-// load config if the file exists
-if (file_exists("src/config.php")) {
-    require_once "src/config.php";
-}
+// load .env
+$dotenv = \Dotenv\Dotenv::createImmutable(dirname(__DIR__,1));
```
Comment (DenverCoder1) [inline tests/StatsTest.php:9]:
```suggestion
$dotenv = \Dotenv\Dotenv::createImmutable(dirname(__DIR__, 1));
```

File: README.md [hunk 5]
```diff
@@ -226,12 +226,11 @@ To get the GitHub API to run locally you will need to provide a token.
 
 1. Visit [this link](https://github.com/settings/tokens/new?description=GitHub%20Readme%20Streak%20Stats) to create a new Personal Access Token
 2. Scroll to the bottom and click **"Generate token"**
-3. **Create a file** `config.php` in the `src` directory and replace `ghp_example123` with **your token** and `DenverCoder1` with **your username**:
+3. **Copy and paste** `.env.example` in the root `.env` directory and replace `TOKEN=` with **your token** and `USERNAME=` with **your username**:
```
Comment (DenverCoder1) [inline README.md:229]:
```suggestion
3. **Make a copy** of `.env.example` named `.env` in the root directory and add **your token** after `TOKEN=` and **your username** after `USERNAME=`
```

**Review by DenverCoder1 — COMMENTED** [review]
Looks good. Just some minor changes.

**Comment by komen205** [comment]
Done @DenverCoder1

File: README.md [hunk 6]
```diff
@@ -226,12 +226,12 @@ To get the GitHub API to run locally you will need to provide a token.
 
 1. Visit [this link](https://github.com/settings/tokens/new?description=GitHub%20Readme%20Streak%20Stats) to create a new Personal Access Token
 2. Scroll to the bottom and click **"Generate token"**
-3. **Create a file** `config.php` in the `src` directory and replace `ghp_example123` with **your token** and `DenverCoder1` with **your username**:
+3. **Make a copy** of `.env.example` named `.env` in the root directory and **your token** after `TOKEN=` and **your username** after `USERNAME=`
+
```
Comment (DenverCoder1) [inline README.md:230]: Forgot the word "add"...

```suggestion
3. **Make a copy** of `.env.example` named `.env` in the root directory and add **your token** after `TOKEN=` and **your username** after `USERNAME=`:
```

**Review by DenverCoder1 — APPROVED** [review]
Thanks for contributing! :tada:

**Pull request merged**